client/src/main/java/io/confluent/kafka/schemaregistry/avro/AvroSchemaUtils.java

```diff
@@ -77,22 +97,64 @@ public static Schema getSchema(Object object, boolean useReflection) {
       return primitiveSchemas.get("Double");
     } else if (object instanceof CharSequence) {
       return primitiveSchemas.get("String");
-    } else if (object instanceof byte[]) {
+    } else if (object instanceof byte[] || object instanceof ByteBuffer) {
       return primitiveSchemas.get("Bytes");
     } else if (useReflection) {
       Schema schema = ReflectData.get().getSchema(object.getClass());
       if (schema == null) {
-        throw new SerializationException("Schema is null for object of class "
-            + object.getClass().getCanonicalName());
+        throw new SerializationException("Schema is null for object of class " + object.getClass()
+            .getCanonicalName());
       } else {
         return schema;
       }
     } else if (object instanceof GenericContainer) {
       return ((GenericContainer) object).getSchema();
+    } else if (object instanceof Map) {
+      // This case is unusual -- the schema isn't available directly anywhere, instead we have to
+      // take get the value schema out of one of the entries and then construct the full schema.
+      Map mapValue = ((Map) object);
+      if (mapValue.isEmpty()) {
+        // In this case the value schema doesn't matter since there is no content anyway. This
+        // only works because we know in this case that we are only using this for conversion and
+        // no data will be added to the map.
+        return Schema.createMap(primitiveSchemas.get("Null"));
+      }
+      Schema valueSchema = getSchema(mapValue.values().iterator().next());
+      return Schema.createMap(valueSchema);
     } else {
       throw new IllegalArgumentException(
           "Unsupported Avro type. Supported types are null, Boolean, Integer, Long, "
               + "Float, Double, String, byte[] and IndexedRecord");
     }
   }
+
+  public static Object toObject(JsonNode value, ParsedSchema parsedSchema) throws IOException {
+    Schema schema = ((AvroSchema) parsedSchema).rawSchema();
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    jsonMapper.writeValue(out, value);
+    DatumReader<Object> reader = new GenericDatumReader<Object>(schema);
+    Object object = reader.read(
+        null,
+        decoderFactory.jsonDecoder(schema, new ByteArrayInputStream(out.toByteArray()))
+    );
+    out.close();
```

Comment:
Does this execute in case of an exception ?

Reply:
It's an in-memory stream, so not as critical to close, but anyway I fixed it.  Thanks!